### DISCUSSION TITLE
Why do I get a "useForm is not a function" error when using React Hook forms in a Next JS project?

### DISCUSSION THREAD

**Discussion by s1egh4rdt** [react-hook-form/react-hook-form#11282] [Q&A]
Hello, I encountered a problem when using React Hook Forms in my Next JS project.

Created a separate file to export the form - LoginForm.tsx
```typescript
import type {NextPage} from 'next'
import { useForm, SubmitHandler } from "react-hook-form";
import styles from './Auth.module.scss';

interface IFormInput {
  email: string
  password: string
}

export const LoginForm: NextPage = () => {
  const { register, handleSubmit } = useForm<IFormInput>();
  const onSubmit: SubmitHandler<IFormInput> = (data) => console.log(data);

  return (
    <div className={styles.formBlock}>
      <h1 className={styles.title}>Login</h1>
      <div className={styles.form_container}>
        <form onSubmit={handleSubmit(onSubmit)}>
            <label htmlFor="email">E-mail</label>
            <input type="email" placeholder="E-mail" 
              {...register('email', {
                required: "Email is required.",
                pattern: {
                  value: /^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$/,
                  message: "Email must be correctly"
                }
              })}
            />
            <label htmlFor="password">Password</label>
            <input type="password"  
              {...register('password', {
                required: "Password is required..",
              })}
              />
            <input type="submit" />
        </form>
      </div>
    </div>
  )
}
``` 

I get an error: Error: (0 , react_hook_form__WEBPACK_IMPORTED_MODULE_1__.useForm) is not a function
![2023-12-11_13-35-26](https://github.com/react-hook-form/react-hook-form/assets/19279983/fd120a95-4676-4071-a649-9a2eb0182041)

I can't figure out what the problem is. Why useForm doesn't work?
I have tried to uninstall the node_modules directory. None of these worked.

**Comment by mattsnow** [comment]
Hi, just to add to this I have the exact same problem. 

I've created a brand new Next JS project and installed React Hook Form. I've copied the default example for Hook Form and nothing else and get this error when rendering: 

`Attempted import error: 'useForm' is not exported from 'react-hook-form' (imported as 'useForm').`

Which then leads to the same error described by OP: 

`TypeError: (0 , react_hook_form__WEBPACK_IMPORTED_MODULE_1__.useForm) is not a function.`

This is driving me crazy as I've got a blank, brand new project with everything up to date and I can't even get the example to render. Any ideas? My package.jsoon is below for reference. 

```
{
  "name": "testing",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "next": "14.0.4",
    "react": "^18",
    "react-dom": "^18",
    "react-hook-form": "^7.49.2"
  },
  "devDependencies": {
    "@types/node": "^20",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "autoprefixer": "^10.0.1",
    "eslint": "^8",
    "eslint-config-next": "14.0.4",
    "postcss": "^8",
    "tailwindcss": "^3.3.0",
    "typescript": "^5"
  }
}
```

**Comment by HugoQueva** [comment]
Have you made sure your `page.tsx` file or `component` has the **"use client"** tag at the top of it?

**Reply by mattsnow** [reply]
This was exactly it! Thank you so much, quite new to react here...

**Reply by tgandee79** [reply]
I've added the ` 'use client'; ` line to the top of the component and I continue to get the same error.  I've tried removing my node_modules and reinstalling them, same problem.

**Reply by Moshyfawn** [reply]
> I've added the `'use client';` line to the top of the component and I continue to get the same error. I've tried removing my node_modules and reinstalling them, same problem.

Take a look at [this Next.js codesandbox](https://codesandbox.io/p/devbox/zen-mclean-xd9lqp?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clqymb5k400073b6fi36mt241%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clqymb5k400023b6fpvs26spb%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clqymb5k400043b6fg2p5e6mb%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clqymb5k400063b6fin1w2q3z%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B48.017530528384256%252C51.982469471615744%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clqymb5k400023b6fpvs26spb%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clqymb5k400013b6fyw44tm10%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%257D%255D%252C%2522id%2522%253A%2522clqymb5k400023b6fpvs26spb%2522%252C%2522activeTabId%2522%253A%2522clqymb5k400013b6fyw44tm10%2522%257D%252C%2522clqymb5k400063b6fin1w2q3z%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clqymb5k400053b6f5oh2na1k%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clqymb5k400063b6fin1w2q3z%2522%252C%2522activeTabId%2522%253A%2522clqymb5k400053b6f5oh2na1k%2522%257D%252C%2522clqymb5k400043b6fg2p5e6mb%2522%253A%257B%2522id%2522%253A%2522clqymb5k400043b6fg2p5e6mb%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clqymb5k400033b6f7garxuj2%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clqymb5k400033b6f7garxuj2%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D). If you remove the `"use client";` directive from the beginning of the file, you will receive this exact error. There are few other possible causes, except for perhaps a typo. I am not questioning you, but without a reproduction, our guess is as good as anyone else's 😉

**Reply by tgandee79** [reply]
I'm an idiot.  I don't know where I got the idea to write "use-client", but that's what I wrote.   I don't know how I didn't see it.  I appreciate the response.   Back to free-time learning.

**Reply by Moshyfawn** [reply]
> I'm an idiot.  I don't know where I got the idea to write "use-client", but that's what I wrote.   I don't know how I didn't see it.  I appreciate the response.   Back to free-time learning.  

Haha, I actually made the exact same mistake when I came back to Next.js at one point, and I'm trying to remember what kind of convention or whatever it is that made me add `"-"` to it, but I can't 🤦

**Reply by devDanielCespedes** [reply]
I love you

**Reply by Pandey-Siddharth** [reply]
I also did the same thing and was trying to solve it since past 30 minutes
